upup/pkg/fi/cloudup/openstacktasks/instance.go

```diff
@@ -151,12 +179,23 @@ func (_ *Instance) ShouldCreate(a, e, changes *Instance) (bool, error) {
 	return a == nil, nil
 }
 
+func makeServerName(e *Instance) string {
+	hash := utils.RandomString(6)
+	// FIXME: Must ensure 63 or less characters
+	// replace all dots and _ with -, this is needed to get external cloudprovider working
+	stripped := strings.Replace(fi.StringValue(e.ClusterName), ".k8s.local", "", -1)
+	iName := strings.Replace(strings.ToLower(fmt.Sprintf("%s-%s.%s", fi.StringValue(e.GroupName), hash, stripped)), "_", "-", -1)
```

Comment:
Possibly using [strings.Replacer](https://pkg.go.dev/strings?tab=doc#Replacer) makes the code more clear and removes the need for a temporary variable. As this is the second occurrence where we need that logic we could create a package var for the replacer.